instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware.py

```diff
@@ -88,6 +88,23 @@ def _get_span_name(request):
         except Resolver404:
             return "HTTP {}".format(request.method)
 
+    @staticmethod
+    def _set_http_route(span, request):
+        # Use the resolved fuction path for the http.route
+        # Note: resolved_match is not available during the process_request phase.
+        if request.resolver_match is not None:
+            func = request.resolver_match.func
+            func_path = None
+            if not hasattr(func, "__name__"):
+                # A class-based view
+                func_path = (
+                    func.__class__.__module__ + "." + func.__class__.__name__
+                )
+            else:
+                # A function-based view
+                func_path = func.__module__ + "." + func.__name__
+            span.set_attribute("http.route", func_path)
```

Comment:
`http.route` is supposed to be a route (usually templated or named) from an HTTP router. In django's case, that would be the string representation of a URL from urls.py. Looks like we are using the Python function/handler name. I'm not sure how useful this tag is but irrespective of whether we add it or not, it shouldn't be named `http.route` as it means something else entirely unless I'm mistaken.
Spec: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions

Reply:
What we want is the `http.route` to be in the span attribute. But you are correct, the value should be populated from the request path (which will be a string from urls.py). I believe it should be within the `request` parameter.

Reply:
See [Opencensus](https://github.com/census-instrumentation/opencensus-python/blob/master/contrib/opencensus-ext-django/opencensus/ext/django/middleware.py#L213

Reply:
@owais / @lzchen Thank you for you comments.

I chose to use class/function name that the request is being routed to so that you would be able to go directly from the http.route as reported in AppInsights and other tools) to the code.  But I do see that this is not in accordance with the spec so I will make the change so the http.route is set to the "matched route (path template)".   

However, I do not agree that the "matched route (path template)" is the request.path as opencensus uses.  This is because the request.path is the actual path of the request, not the route or path template.  One path template, you could see multiple paths because the parameters in the path will vary.  For instance, the following paths from `request.path`:
/api/1/financial-institutions/111/plaid_details/
/api/1/financial-institutions/444/plaid_details/
/api/1/financial-institutions/554/plaid_details/
would all map to one route:
^api/1/financial-institutions/(?P<pk>[^/.]+)/plaid_details/$'

I have only been able to find this route as being available from `request.resolver_match.route` and `request.resolver_match` is none during `process_request()` but become available during `process_response` and `process_exception`.  So I think this value cannot be set just in `process_request()`.

I will go ahead and make the change to setting http.route to request.resolver_match.route and push up the change.

Please let me know what you think.  Thanks!

Reply:
> However, I do not agree that the "matched route (path template)" is the request.path as opencensus uses.

If you are trying to add this to enable your AppInsights scenario, I suggest doing it the same way OpenCensus is doing it. OpenCensus Python SDK is Microsoft's current solution for Python SDK to send telemetry to App Insights and so being consistent with it is important for back compat and to not break users. If you want to propose a new design, I think submitting a separate feature request (after merging this PR) would be good.

Reply:
@lzchen - I have change the code so that it used request.path for the http.route value.

Reply:
I think we should use templated path/route for this and stick with OpenTelemetry spec. If we decide to have same behavior of OC, we'll have same information duplicated in `http.target` and `http.route` attributes. I don't think (may be I'm wrong) the breakage here is big enough to warrant going against Otel semantic conventions. OpenCensus users can still find their old value for `http.route` in OpenTelemetry's `http.target` so no data is lost after the migration, only renamed. WDYT?

Reply:
@owais 
I thought request.path represented the templated path/route, but if it isn't then that is my misunderstanding. But yes, I agree with what you are saying.

Reply:
Glad we are arriving at a consensus.  I have switched the code back to using request.resolver_match.route which is the templated route.  I also realized from looking at the OpenCensus implementation that though resolver_match is not set for process_request, it is available in process_view so I added that method.

Also http.path was being set in the OpenCensus implementation by not in the OpenTelemetry implementation so I also added that.